src/platform/device-manager.ts

```diff
@@ -3,23 +3,81 @@
  * TODO(cjdaly) integrate with Mock Local Home Platform and Mock Network
  **/
 /// <reference types="@google/local-home-sdk" />
+export const ERROR_UNEXPECTED_COMMAND_REQUEST =
+  'Unable to process unexpected CommandRequest';
+export class DeviceManagerStub implements smarthome.DeviceManager {
+  /** Action to call when an `IntentRequest` is marked with `markPending()` */
+  private markPendingAction:
+    | ((request: smarthome.IntentRequest) => void)
+    | undefined;
+  /** Map of each expected `CommandRequest` to its associated response*/
+  private expectedCommandToResponse: Map<
+    smarthome.DataFlow.CommandRequest,
+    smarthome.DataFlow.CommandBase
+  > = new Map();
 
-import { DeviceManager } from '@google/local-home-sdk';
+  /**
+   * Checks if the next request marked pending matches a given request.
+   * @param requestToMatch  A request to test against the next request marked pending.
+   * @returns  Promise that resolves to a boolean representing if the requests matched.
+   */
+  public doesNextPendingRequestMatch(
+    requestToMatch: smarthome.IntentRequest
+  ): Promise<boolean> {
+    return new Promise<boolean>(resolve => {
+      this.markPendingAction = (pendingRequest: smarthome.IntentRequest) => {
+        resolve(pendingRequest === requestToMatch);
+      };
+    });
+  }
+
+  /**
+   * Registers a command that will be checked against on `send()`, and a corresponding
+   * response that will be returned from `send()` on a successful match.
+   * @param expectedCommand  The command to check against incoming commands.
+   * @param response  The response to send when an incoming command matches.
+   */
+  public addExpectedCommand(
```

Comment:
I'm not really sure what you mean.  If we buffer the `DeviceManager` commands and verify them after they are sent, how would we mock the responses per-command? (Developers may want to test different scenarios with `CommandSuccess` and `CommandFailue` responses depending on the command)

Reply:
I guess it might help is to buffer the sent commands anyways.  In case a command doesn't match, it would allow the developer to see what commands were sent vs. what was expected.

Reply:
I added a buffer to store sent commands and some assertion helpers.  I updated the tests to use them.